tangelo/toolboxes/post_processing/histogram.py

```diff
@@ -0,0 +1,225 @@
+        return f"{self.counts}"
+
+    def __eq__(self, other):
+        """Two histograms are equal if they have same counts, i.e. same
+        bistrings and outcomes.
+        """
+        return (self.counts == other.counts)
+
+    def __add__(self, other):
+        """The aggregate_histograms function is used to add two Histogram objects."""
+        return aggregate_histograms(self, other)
+
+    def __iadd__(self, other):
+        new_histogram = self + other
+        self.__dict__ = new_histogram.__dict__
+        return self
+
+    @property
+    def n_shots(self):
+        """Return the number of shots encoded in the Histogram.
+
+        Returns:
+            int: Self-explanatory.
+        """
+        return sum(self.counts.values())
+
+    @property
+    def n_qubits(self):
+        """The length of the bistrings represents the number of qubits."
+
+        Returns:
+            int: Self-explanatory.
+        """
+        return len(list(self.counts.keys())[0])
+
+    @property
+    def frequencies(self):
+        """The frequencies are normalized counts vs the number of shots.
+
+        Returns:
+            dict: Frequencies in a {bistring: probability, ...} format.
+        """
+        return {bistring: counts/self.n_shots for bistring, counts in self.counts.items()}
+
+    def remove_bits(self, *indices):
+        """Method to remove bits on given indices.
+
+        Args:
+            indices (variable number of int): Qubit indices to remove.
+        """
+        new_counts = dict()
+        for bitstring, counts in self.counts.items():
+            new_bistring = "".join([bitstring[qubit_i] for qubit_i in range(len(bitstring)) if qubit_i not in indices])
+
+            if new_bistring in new_counts:
+                new_counts[new_bistring] += counts
+            else:
+                new_counts[new_bistring] = counts
+
+        self.counts = new_counts
+
+    def post_select(self, expected_outcomes):
+        """Method to apply post selection on Histogram data, based on a post
+        selection function in this module. This method change the Histogram
+        object inplace.
+
+        Args:
+            expected_outcomes (dict): Wanted outcomes on certain qubit indices and
+                their expected state. For example, {0: "1"} would filter results
+                based on the first qubit with the 1 state measured. This argument
+                can also filter many qubits.
+        """
+        def f_post_select(bitstring):
+            for qubit_i, expected_bit in expected_outcomes.items():
+                if bitstring[qubit_i] != expected_bit:
+                    return False
+            return True
+
+        new_hist = filter_hist(self, f_post_select)
+        self.counts = new_hist.counts
+        self.remove_bits(*list(expected_outcomes.keys()))
+
+    def resample(self, n_shots):
+        """Post selection is done with the tangelo.toolboxes.post_processing.boostrapping.get_resampled_frequencies
```

Comment:
I'd avoid naming the module etc because we need to change this if the function is renamed or moved.
You already have an import in the file, the user is a big boy: they can figure it out.

Further more this docstring is incorrect: this is not about post selection. Simply state that you are generating a new histogram with n_shots counts, based on the distribution given by the input Histogram object.